dianna/methods/kernelshap.py

```diff
@@ -147,6 +149,8 @@ def _prepare_image_data(self, input_data):
         Returns:
             transformed input data
         """
+        if self.preprocess_function is not None:
```

Comment:
I moved the `preprocess_function` to the function `_prepare_image_data`. There the `preprocess_function` is executed before the processing of input data inside `kernelshap` and therefore the user does not need to be bothered by the operations inside `kernelshap` when implementing their preferred way of preprocessing the input image. I think this is more friendly to the user.

Reply:
I'm not sure I get why you moved it all the way to the start of the processing. What would then be the point of having a preprocessing function? The user could then just do it themselves before using dianna. 
In the other methods, the preprocess function is called as late as possible, just before running the model. The reason being that the model might require slightly different input than the XAI method. E.g. Images must have a channels axis, which you might not have for greyscale data. Then you could add the channels axis, and add a preprocessing function that removes the channels axis. Such a preprocessing  function should only be called just before running the model. 
You're absolutely right that it requires knowledge of the internals of the method, but I think that is exactly the point in this case

Reply:
Hi @loostrum , thanks for your explanation. My impression about `preprocess` comes from some other python packages (for instance, like [xarray.open_mfdataset](https://xarray.pydata.org/en/stable/generated/xarray.open_mfdataset.html). The process begins before the operations regarding the input. But I totally agree with you in this case and I think what you suggest here make more sense. Thanks again for the clarification and sorry for making this quite confusing. I will revert my change and just go ahead and merge it.

Reply:
No problem, thanks for picking up this PR again!